cpp/Esri/ArcGISRuntime/Toolkit/CoordinateConversionController.h

```diff
@@ -0,0 +1,309 @@
+ *   - The controller to zoom to the point's location on the GeoView. 
+ *   - The user to select the point with a mouse-event on the GeoView. 
+ *   - The controller to calculate the screen coordinate of the current point
+ *     relative to the window containing the GeoView.
+ * 
+ */
+class CoordinateConversionController : public QObject
+{
+  Q_OBJECT
+  Q_PROPERTY(QObject* geoView READ geoView WRITE setGeoView NOTIFY geoViewChanged)
+  Q_PROPERTY(double zoomToDistance READ zoomToDistance WRITE setZoomToDistance NOTIFY zoomToDistanceChanged)
+  Q_PROPERTY(QAbstractListModel* formats READ coordinateFormats CONSTANT)
+  Q_PROPERTY(QAbstractListModel* results READ conversionResults CONSTANT)
+  Q_PROPERTY(bool inPickingMode READ inPickingMode WRITE setInPickingMode NOTIFY inPickingModeChanged)
+public:
+  /*!
+   * \brief Constructor.
+   * \param parent owning parent object.
+   */
+  Q_INVOKABLE CoordinateConversionController(QObject* parent = nullptr);
+
+  /*!
+   * \brief Destructor
+   */
+  ~CoordinateConversionController() override;
+
+  /*!
+   * \brief Returns the GeoView as a QObject.
+   * \return The current GeoView as a QObject.
+   */
+  QObject* geoView() const;
+
+  /*!
+   * \brief Set the GeoView object this Controller uses.
+   * 
+   * Internally this is cast to a MapView or SceneView using `qobject_cast`,
+   * which is why the paremeter is of form `QObject` and not `GeoView`.
+   * 
+   * \param geoView Object which must inherit from GeoView* and QObject*. 
+   */
+  void setGeoView(QObject* mapView);
+
+  /*!
+   * \brief Known list of available coordinate conversion formats which can be
+   * consumed to generate different textual representations of the same point.
+   * Internally, this is a GenericListModel with an elementType of
+   * CoordinateConversionOption .
+   * 
+   * \sa GenericListModel
+   * 
+   * \return GenericListModel of CoordinateConversionOption elements.
+   */
+  GenericListModel* coordinateFormats() const;
+
+  /*!
+   * \brief List of textual representations of the current point in different 
+   * formats.
+   * Internally, this is a GenericListModel with an elementType of
+   * CoordinateConversionResult.
+   * 
+   * \sa GenericListModel
+   * 
+   * \return GenericListModel of CoordinateConversionResult elements.
+   */
+  GenericListModel* conversionResults() const;
+
+  /*!
+   * \brief Returns the current point.
+   * 
+   * The current point represents the one point that all elements within
+   * the conversionResults list-model are tied to.
+   * 
+   * \return Point 
+   */
+  Point currentPoint() const;
+
+  /*!
+   * \brief Converts the current point held by this controller as a 2D point 
+   * relative to the current window.
+   * 
+   * \return The point returned by currentPoint as a screen coordinate. 
+   */
+  Q_INVOKABLE QPointF screenCoordinate() const;
+
+  /*!
+   * \brief Distance between camera and current point when zoomToCurrentPoint
```

Comment:
in meters (?)

Reply:
I have commented further down that the units are those defined as distance in the geoView's SpatialReference.